### ISSUE TITLE
Transformations available on 'ValueChoice' 

### ISSUE THREAD

**Issue by datapaintings** [microsoft/nni#3571]
I would like to search for multiinput neural networks with tuned hidden size of each mode, f. e.:

        self.hidden_size_1 = nn.ValueChoice([16, 32, 64, 128, 256, 512], label='hidden_size_1')
        self.hidden_size_2 = nn.ValueChoice([16, 32, 64, 128, 256, 512], label='hidden_size_2')
        self.output_layer = nn.Linear(self.hidden_size_1 + self.hidden_size_2, 2)

Currently, due to documentation, transformations on 'ValueChoice' are not supported, so concatenation of modes cannot be performed.

**Comment by ultmaster** [comment]
I think it is possible but requires quite a lot of implementation effort. So IMHO, we should put that in backlog.
